restapi/src/main/java/io/stargate/web/docsapi/service/query/search/db/impl/FilterPathSearchQueryBuilder.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.web.docsapi.service.query.search.db.impl;
+
+import io.stargate.db.query.Predicate;
+import io.stargate.db.query.builder.BuiltCondition;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import io.stargate.web.docsapi.service.query.DocumentServiceUtils;
+import io.stargate.web.docsapi.service.query.FilterPath;
+import io.stargate.web.docsapi.service.query.QueryConstants;
+import io.stargate.web.docsapi.service.query.search.db.AbstractSearchQueryBuilder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** The search query builder that creates all needed predicates for a {@link FilterPath}. */
+public class FilterPathSearchQueryBuilder extends AbstractSearchQueryBuilder {
+
+  private final FilterPath filterPath;
+
+  private final boolean matchField;
+
+  /**
+   * @param filterPath Filter path
+   * @param matchField If field name should be matches as well, adds extra predicates
+   */
+  public FilterPathSearchQueryBuilder(FilterPath filterPath, boolean matchField) {
+    this.filterPath = filterPath;
+    this.matchField = matchField;
+  }
+
+  @Override
+  protected Map<String, Predicate> getBindPredicates() {
+    return Collections.emptyMap();
+  }
+
+  @Override
+  public Collection<BuiltCondition> getPredicates() {
+    List<BuiltCondition> predicates = getPathPredicates();
+    predicates.addAll(getFieldPredicates());
+    predicates.addAll(getRemainingPathPredicates());
```

Comment:
Nope, because each of those method is complicated enough to be separated into single concern. Having the complete code in one method is actually decreasing the clarity imo.

Reply:
:+1: